src/bugsnag.js

```diff
@@ -314,34 +318,56 @@
       });
     }
 
-    enhance(console, "log", function() {
-      trackLog("log", arguments);
-    });
+    // feature detection for console.log
+    if(typeof window.console === "undefined") {
+      return;
+    }
 
-    enhance(console, "warn", function() {
-      trackLog("warn", arguments);
-    });
+    // keep track of functions that we will need to hijack
+    var originalLog, originalWarn, originalError;
 
-    enhance(console, "error", function() {
-      trackLog("error", arguments);
-    });
+    self.enableAutoBreadcrumbsConsole = function() {
+      self.autoBreadcrumbsConsole = true;
+
+      originalLog = console.log;
+      originalWarn = console.warn;
+      originalError = console.error;
+
+      enhance(console, "log", function() {
+        trackLog("log", arguments);
+      });
+
+      enhance(console, "warn", function() {
+        trackLog("warn", arguments);
+      });
+
+      enhance(console, "error", function() {
+        trackLog("error", arguments);
+      });
+    };
+
+    self.disableAutoBreadcrumbsConsole = function() {
+      self.autoBreadcrumbsConsole = false;
+
+      console.log = originalLog;
+      console.warn = originalWarn;
+      console.error = originalError;
```

Comment:
This could be bad in a case where someone calls one of the disable functions before enable has been called. Maybe we could handle that case with something like:

```javascript
 console.log = originalLog || console.log;
```